### PR TITLE
Add dockerlint

### PR DESCRIPTION
**Pull request by RDIL**
## PR Summary

Adds the dockerlint tool

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `Dockerfile`, `.sh`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **Adds a new image or tag**
  - [x] Not Applicable
- **OR**
  - [ ] Update [README.powershellcommunity.md](https://github.com/PowerShell/PowerShell-Docker/blob/master/assets/README.powershellcommunity.md)
  - [x] Update [vsts-ci.yml](https://github.com/PowerShell/PowerShell-Docker/blob/master/vsts-ci.yml)

and also since we are in the middle of a pandemic, here is something to cheer everyone up
Screenshot 2020-05-06 at 9 08 22 AM

### CODE REVIEW

File: .vsts-ci/phase.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 parameters:
   name: ''
-  vmImage: 'ubuntu-16.04'
+  vmImage: 'ubuntu-18.04'
```
Comment (RDIL) [inline .vsts-ci/phase.yml:3]: Had to bump this up because Xenial doesn't have the needed GLIBC installed

**Comment by RDIL** [comment]
Hey @TravisEz13 do you know how we can get Start-Process to output to stdout?

**Comment by TravisEz13** [comment]
You should use this wrapper to execute native commands:  https://github.com/PowerShell/PowerShell/blob/b7c66ab3ef584956a2872856ee8b351784f522d9/build.psm1#L1396

**Comment by RDIL** [comment]
That's only for windows, plus I changed it to just use bash.

**Comment by TravisEz13** [comment]
You should really create a python package.  Put it in the GitHub python package registry:  https://github.com/features/packages

Then do the rest of the steps on the CI machine.

**Comment by RDIL** [comment]
All the CI machine is doing right now is downloading and decompressing the executable...

plus it already a python package that I'm building into an executable.  I suppose I could setup the CI to clone the repo or something but I feel like that wouldn't be an efficient way to do it

**Comment by TravisEz13** [comment]
But building it to a native on the other machine, causes the wrong libc to be needed.

**Comment by RDIL** [comment]
Fair enough, should be fixed now

**Comment by RDIL** [comment]
Its probably because its built natively on a python docker image, so that issue should be fixed now.

**Comment by RDIL** [comment]
@TravisEz13 good news: its working. Bad news: it isn't working on windows so does azure devops have an environment variable for if it is on windows so I can skip

File: .vsts-ci/phase.yml [hunk 2]
```diff
@@ -55,11 +56,11 @@ jobs:
       make
       cd ..
     displayName: Download dockerlint
-    condition: succeeded()
+    condition: ne(variables['AGENT_OS'], 'Windows_NT')
```
Comment (TravisEz13) [inline .vsts-ci/phase.yml:59]:
```suggestion
    condition: and( succeeded(), ne(variables['AGENT_OS'], 'Windows_NT'))
```

File: .vsts-ci/phase.yml [hunk 3]
```diff
@@ -55,11 +56,11 @@ jobs:
...
 
   - bash: python3 dockerlint/main.py -d release/${CHANNEL}/${IMAGENAME}/docker/Dockerfile
     displayName: Run dockerlint
-    condition: succeeded()
+    condition: ne(variables['AGENT_OS'], 'Windows_NT')
```
Comment (TravisEz13) [inline .vsts-ci/phase.yml:63]:
```suggestion
    condition: and( succeeded(), ne(variables['AGENT_OS'], 'Windows_NT'))
```

**Comment by RDIL** [comment]
@TravisEz13 any idea why this is still running on windows

File: .vsts-ci/phase.yml (hunk 1 above)
Comment (TravisEz13) [inline .vsts-ci/phase.yml:3]:
```suggestion
  pool:
    vmImage: 'ubuntu-18.04'
```

**Comment by TravisEz13** [comment]
Can you add a failure to one of the images and let me see what it looks like?

**Comment by RDIL** [comment]
The upgrade to the beta broke the failure reporting, working on fixing it now.

**Comment by RDIL** [comment]
@TravisEz13 see https://dev.azure.com/powershell/docker/_build/results?buildId=51668&view=logs&j=59760ad3-d83c-5873-e7bc-29164ae56e48&t=a3ff8f5a-08da-5f2f-269f-a49fac996a03&l=21 for example failure

**Comment by RDIL** [comment]
Fyi @TravisEz13 here is example output for the script:
```powershell
/bin/bash --noprofile --norc /home/vsts/work/_temp/529d60fd-1be4-4387-b296-097b95f45f9e.sh

    Starting dockerlint...

info: Using dockerfile from path: release/stable/debian9/docker/Dockerfile
issue: line 20 - uses apt install without no-install-recommends lacks-no-install-recommends
issue: line 33 - uses apt install without no-install-recommends lacks-no-install-recommends
```

Meaning that issue lines follow the format of:
`issue: line [NUMBER] - [DESCRIPTION] [PROBLEM-ID]`

**Comment by TravisEz13** [comment]
Would be nice if it showed up as a test, but this is fine.

**Comment by TravisEz13** [comment]
but we can't commit it broken...

**Comment by RDIL** [comment]
Yeah, I was going to fix it later today.  What test result format does it need to output for it to show up in azure devops?

**Comment by TravisEz13** [comment]
https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/publish-test-results?view=azure-devops&tabs=yaml
> JUnit, NUnit, VSTest, xUnit, cTest
One of those

**Comment by RDIL** [comment]
Alright, I'll implement reporting through JUnit-compatible XML data.

**Comment by RDIL** [comment]
@PoshChan please rerun CI

**Comment by PoshChan** [comment]
@RDIL, successfully started rebuild of `docker-PR`

**Comment by TravisEz13** [comment]
test results look good

**Comment by TravisEz13** [comment]
thanks

**Comment by RDIL** [comment]
Alright, merging the junit feature into master, then will fix the issues here.

**Comment by TravisEz13** [comment]
Thanks for your contribution

**Comment by RDIL** [comment]
@TravisEz13 do you have any suggestions for checks I should add?

**Comment by TravisEz13** [comment]
Not right now, in the future, we want the from tags to be static (not a varible)

**Pull request merged**